File: examples/go/lmstream/enums.go
package lmstream

type Opcode uint8

const (
	OpcodeOpen      Opcode = 0x01
	OpcodeOpenAck   Opcode = 0x02
	OpcodeContent   Opcode = 0x03
	OpcodeReasoning Opcode = 0x04
	OpcodeToolCall  Opcode = 0x05
	OpcodeUsage     Opcode = 0x06
	OpcodeMetrics   Opcode = 0x07
	OpcodeEnd       Opcode = 0x08
	OpcodeError     Opcode = 0x09
	OpcodeCancel    Opcode = 0x0A
	OpcodePause     Opcode = 0x0B
	OpcodeResume    Opcode = 0x0C
	OpcodeSteer     Opcode = 0x0D
	OpcodeRewind    Opcode = 0x0E
	OpcodeSettings  Opcode = 0x0F
	OpcodePing      Opcode = 0x10
	OpcodePong      Opcode = 0x11
)

const (
	HeaderFlagBoundary     uint8 = 0x01
	HeaderFlagCompressed   uint8 = 0x02
	HeaderFlagReservedMask uint8 = 0xFE
)

const (
	ToolFlagStart        uint8 = 0x01
	ToolFlagCont         uint8 = 0x02
	ToolFlagEnd          uint8 = 0x04
	ToolFlagReservedMask uint8 = 0xF8
)

type FinishReason uint8

const (
	FinishReasonStop          FinishReason = 0x00
	FinishReasonLength        FinishReason = 0x01
	FinishReasonToolCalls     FinishReason = 0x02
	FinishReasonContentFilter FinishReason = 0x03
	FinishReasonCancelled     FinishReason = 0x04
)

type ErrorCode uint16

const (
	ErrorCodeBadRequest        ErrorCode = 1001
	ErrorCodeAuthFailed        ErrorCode = 1002
	ErrorCodeRateLimited       ErrorCode = 1003
	ErrorCodeModelOverloaded   ErrorCode = 1004
	ErrorCodeUpstreamError     ErrorCode = 1005
	ErrorCodeBufferOverflow    ErrorCode = 1006
	ErrorCodeInvalidState      ErrorCode = 1007
	ErrorCodeProtocolViolation ErrorCode = 1008
	ErrorCodeSessionExpired    ErrorCode = 1009
)

type ErrorScope uint8

const (
	ErrorScopeSession    ErrorScope = 0x00
	ErrorScopeConnection ErrorScope = 0x01
)

File: examples/python/lmstream/enums.py
from enum import IntEnum

class Opcode(IntEnum):
    OPEN = 0x01
    OPEN_ACK = 0x02
    CONTENT = 0x03
    REASONING = 0x04
    TOOL_CALL = 0x05
    USAGE = 0x06
    METRICS = 0x07
    END = 0x08
    ERROR = 0x09
    CANCEL = 0x0A
    PAUSE = 0x0B
    RESUME = 0x0C
    STEER = 0x0D
    REWIND = 0x0E
    SETTINGS = 0x0F
    PING = 0x10
    PONG = 0x11

class HeaderFlags(IntEnum):
    BOUNDARY = 0x01
    COMPRESSED = 0x02
    RESERVED_MASK = 0xFE

class ToolFlags(IntEnum):
    START = 0x01
    CONT = 0x02
    END = 0x04
    RESERVED_MASK = 0xF8

class FinishReason(IntEnum):
    STOP = 0x00
    LENGTH = 0x01
    TOOL_CALLS = 0x02
    CONTENT_FILTER = 0x03
    CANCELLED = 0x04

class ErrorCode(IntEnum):
    BAD_REQUEST = 1001
    AUTH_FAILED = 1002
    RATE_LIMITED = 1003
    MODEL_OVERLOADED = 1004
    UPSTREAM_ERROR = 1005
    BUFFER_OVERFLOW = 1006
    INVALID_STATE = 1007
    PROTOCOL_VIOLATION = 1008
    SESSION_EXPIRED = 1009

class ErrorScope(IntEnum):
    SESSION = 0x00
    CONNECTION = 0x01
